lib/ProgressPlugin.js

```diff
@@ -66,18 +72,28 @@ const createDefaultHandler = profile => {
 };
 
 class ProgressPlugin {
+	/**
+	 * @param {ProgressPluginArgument} options options
+	 */
 	constructor(options) {
 		if (typeof options === "function") {
 			options = {
 				handler: options
 			};
 		}
+
 		options = options || {};
+		validateOptions(schema, options, "Progress Plugin");
+		options = Object.assign(ProgressPlugin.defaultOptions, options);
```

Comment:
`Object.assign({}, ProgressPlugin.defaultOptions, options)`

otherwise you mutate the defaultOptions.